src/lib/visualization/lqp_visualizer.cpp

```diff
@@ -151,35 +151,37 @@ void LQPVisualizer::_build_dataflow(const std::shared_ptr<AbstractLQPNode>& sour
   }
 
   if (!dynamic_pointer_cast<AbstractNonQueryNode>(source_node)) {
-    // Edge Tooltip: Unique Constraints
-    const auto& unique_constraints = source_node->unique_constraints();
+    // Edge Tooltip: Unique Column Combinations.
+    const auto& unique_column_combinations = source_node->unique_column_combinations();
     tooltip_stream << "\n"
-                   << "Unique Constraints: \n";
-    if (unique_constraints->empty()) {
+                   << "Unique Column Combinations: \n";
+    if (unique_column_combinations->empty()) {
       tooltip_stream << " <none>\n";
     }
 
-    const auto unique_constraint_count = unique_constraints->size();
-    for (auto constraint_index = size_t{0}; constraint_index < unique_constraint_count; ++constraint_index) {
-      tooltip_stream << " (" << constraint_index + 1 << ") ";
-      tooltip_stream << unique_constraints->at(constraint_index) << "\n";
+    auto ucc_idx = 0;
+    for (const auto& ucc : *unique_column_combinations) {
+      ++ucc_idx;
+      tooltip_stream << " (" << ucc_idx << ") ";
+      tooltip_stream << ucc << "\n";
     }
 
-    // Edge Tooltip: Trivial FDs
-    auto trivial_fds = std::vector<FunctionalDependency>();
-    if (!unique_constraints->empty()) {
-      trivial_fds = fds_from_unique_constraints(source_node, unique_constraints);
+    // Edge Tooltip: Trivial FDs.
+    auto trivial_fds = FunctionalDependencies();
+    if (!unique_column_combinations->empty()) {
+      trivial_fds = fds_from_unique_column_combinations(source_node, unique_column_combinations);
     }
     tooltip_stream << "\n"
                    << "Functional Dependencies (trivial): \n";
     if (trivial_fds.empty()) {
       tooltip_stream << " <none>\n";
     }
 
-    const auto trivial_fd_count = trivial_fds.size();
-    for (auto fd_idx = size_t{0}; fd_idx < trivial_fd_count; ++fd_idx) {
-      tooltip_stream << " (" << fd_idx + 1 << ") ";
-      tooltip_stream << trivial_fds.at(fd_idx) << "\n";
+    auto trivial_fd_idx = 0;
```

Comment:
Doesn't matter but just curious why you initialize with 0 just to immediately increment it?